### DISCUSSION TITLE
Tree-shaking fails for different imports from the same file across multiple routes

### DISCUSSION THREAD

**Discussion by RhythmAgg** [vercel/next.js#66754] [Help]
### Summary

refer: [Original Issue](https://github.com/vercel/next.js/issues/34559)
I am facing the exact issue in my production build. Since the original issue is closed and is a couple of years old, I would like to know what are the current solutions to the solution. As mentioned in the original discussion by @sokra, creating different modules for each export is difficult in my case given the shared module has inter-dependencies and thousands of exports. 

Also i would like to know why wouldn't `optimizations per runtime` work for this issue. As per the doc:
> Webpack 5 is now able (and does by default) to analyse and optimize modules per runtime (A runtime is often equal to an entrypoint). This allows to only exports in these entrypoints where they are really needed.
***
This should mean for each runtime( entry point ) if we add this optimization our next.config.js the unused exports will be re-calculated for that runtime. Hence for each page only the required code should be bundled. Please help me out understanding this optimization where i am going wrong.

### Example

 https://github.com/joyt/nextjs-tree-shake-fail

**Comment by lykianovsky** [comment]
https://github.com/vercel/next.js/issues/69532

Maybe my solution can help you

**Reply by RhythmAgg** [reply]
Yes so we eventually resort to splitting the utility modules with many exports into smaller modules exporting single items. For this we developed codemods that used jscodeshift AST parser and created the sub modules. These codemods follow the code structure respective to our company.
